feat(docs): Add documentation for EU-specific gated models (#1874)
huggingface/hub-docs: docs/hub/models-gated.md

You can also provide the `token` parameter to most loading methods in the libraries (`from_pretrained`, `hf_hub_download`, `load_dataset`, etc.), directly from your scripts.

For more details about how to login, check out the [login guide](https://huggingface.co/docs/huggingface_hub/quick-start#login).

### Restricting Access for EU Users

> For gated models, you can add an additional layer of access control to specifically restrict users from European Union countries. This is useful if your model's license or terms of use prohibit its distribution in the EU.

Reviewer: ```suggestion
Gated repositories can be configured with an additional layer of access control to disallow access from European Union countries. This is useful if your license or terms of use prohibit distribution in the EU.
```

Not specific to models, I think (even though the rest of file does mention models everywhere).
